apps/dashboard/src/main/java/com/akto/action/DashboardAction.java

```diff
@@ -255,6 +249,35 @@ public String markConnectionAsSkipped(){
         }
     }
 
+    private String email;
+    private String username;
+    private String organization;
+    public String updateUsernameAndOrganization() {
+        if(username.trim().isEmpty()) {
+            addActionError("Username cannot be empty");
+            return Action.ERROR.toUpperCase();
+        }
+        User user = UsersDao.instance.updateOne(Filters.in(User.LOGIN, email), Updates.combine(
+                Updates.set(User.NAME, username.trim()),
+                Updates.set(User.NAME_LAST_UPDATE, Context.now())
+        ));
+        RBAC.Role currentRoleForUser = RBACDao.getCurrentRoleForUser(user.getId(), Context.accountId.get());
+
+        if(currentRoleForUser.equals(RBAC.Role.ADMIN)) {
+            if(organization.trim().isEmpty()) {
+                addActionError("Organization cannot be empty");
+                return Action.ERROR.toUpperCase();
+            }
+
+            OrganizationsDao.instance.updateOne(Filters.in(Organization.ADMIN_EMAIL, email), Updates.combine(
+                    Updates.set(Organization.NAME, organization.trim()),
```

Comment:
in case of multiple admins, we store only 1 admin email, and you have not handled upsert false case so it will create new org